src/store/message-store-level.ts

```diff
@@ -52,69 +62,95 @@ export class MessageStoreLevel implements MessageStore {
     }
   }
 
-
   async close(): Promise<void> {
     await this.db.close();
     await this.index.FLUSH();
   }
 
-  async get(cid: CID): Promise<Message> {
-    const bytes = await this.db.get(cid);
+  async get(cid: CID, ctx: Context): Promise<MessageJson> {
+    const bytes = await this.db.get(cid, ctx);
 
     if (!bytes) {
       return;
     }
 
     const decodedBlock = await block.decode({ bytes, codec: cbor, hasher: sha256 });
 
-    return decodedBlock.value as Message;
-  }
+    const jsonMessage = decodedBlock.value as MessageJson;
+    const { descriptor } = jsonMessage;
 
-  async query(query: any): Promise<Message[]> {
-    const messages: Message[] = [];
+    if (!descriptor.dataCid) {
+      return jsonMessage;
+    }
 
-    // copy the query provided to prevent any mutation
-    const copy: any = { ...query };
-    delete copy.method;
+    const dataDagRoot = await exporter(descriptor.dataCid as CID, this.db);
+    const dataBytes = new Uint8Array(dataDagRoot.size);
+    let offset = 0;
 
-    // parse query into a query that is compatible with the index we're using
-    const indexQueryTerms: string[] = MessageStoreLevel.buildIndexQueryTerms(copy);
-    const { RESULTS: indexResults } = await this.index.QUERY({ AND: indexQueryTerms });
+    for await (const chunk of dataDagRoot.content()) {
+      dataBytes.set(chunk, offset);
+      offset += chunk.length;
+    }
```

Comment:
Consider hiding this magic in a method for readability, perhaps:

```suggestion
    const dataBytes = wait this.getUnixFsDagPbData(descriptor.dataCid);
```

Similarly for line 137 ~ 144 in `put()`

Reply:
i don't know if i'm being consistent in saying this relative to the rest of the codebase but i'll say it anyway! 

it's only a handful of lines and it isn't being used anywhere else, so i think it might be ok here for the time being. I also can't think of a good method name lol

Reply:
Sure, this is just a suggestion. Looks like the code is different now to when I first reviewed it anyway. I'll put it into a method if I come across it in next time and having to spend cycles re-learn the block.